library/kotlin/src/io/envoyproxy/envoymobile/filters/FilterTrailersStatus.kt

```diff
@@ -17,4 +17,14 @@ sealed class FilterTrailersStatus<T : Headers> {
    * is desired.
    */
   class StopIteration<T : Headers>(val trailers: T) : FilterTrailersStatus<T>()
+
+  /**
+   * Resume previously-stopped iteration, possibly forwarding headers and data, if iteration was
+   * previously stopped during an on*Headers or on*Data invocation.
+   *
+   * It is an error to return ResumeIteration if iteration is not currently stopped, and it is
+   * an error to include headers if headers have already been forwarded to the next filter
+   * (i.e. iteration was stopped during an on*Data invocation instead of on*Headers).
+   */
+  class ResumeIteration<T : Headers>(val headers: Headers?, val data: ByteBuffer?, val trailers: T) : FilterTrailersStatus<T>()
```

Comment:
Same comment here regarding behavior in the `nil` case

Reply:
I intentionally introduced a discrepancy between Kotlin and Swift here to demonstrate two potential approaches (we should ultimately pick one). The Swift approach allows unused entities to be defaulted to `nil` and left off of the return status. The Kotlin approach positions the entities in the order in which they'll appear.

I'm open to either and think there's pros to both.

Reply:
I was playing around with the Swift playground, and it turns out that with named parameters, defaulted arguments on enum cases can be left off in any position. And with Kotlin I think we can achieve this with multiple constructors. So in short, we should be able to support `.resumeIteration(headers: myHeaders, data: myData)` seamlessly with simple case of `.resumeIteration(data: myData)`, realizing both the benefits of conceptually consistent ordering and defaulting.